newIDE/app/src/InstancesEditor/InstancesResizer.js

```diff
@@ -20,98 +61,243 @@ export default class InstancesResizer {
+      x: instance.getX(),
+      y: instance.getY(),
+    };
+    return initialPosition;
   }
 
-  _getSizeDeltaY(proportional: boolean, initialSize: Dimension) {
-    if (proportional && Math.abs(this.totalDeltaY) < Math.abs(this.totalDeltaX))
-      return (initialSize.height / initialSize.width) * this.totalDeltaX;
-
-    return this.totalDeltaY;
+  _getOrCreateSelectionAABB(instances: gdInitialInstance[]): Rectangle {
+    let initialSelectionAABB = this._initialSelectionAABB;
+    if (initialSelectionAABB) {
+      return initialSelectionAABB;
+    }
+    initialSelectionAABB = new Rectangle();
+    initialSelectionAABB.setRectangle(
+      this._getOrCreateInstanceAABB(instances[0])
+    );
+    for (let i = 1; i < instances.length; i++) {
+      initialSelectionAABB.union(this._getOrCreateInstanceAABB(instances[i]));
+    }
+    this._initialSelectionAABB = initialSelectionAABB;
+    return initialSelectionAABB;
   }
 
   resizeBy(
     instances: gdInitialInstance[],
     deltaX: number,
     deltaY: number,
+    grabbingLocation: ResizeGrabbingLocation,
     proportional: boolean
   ) {
     this.totalDeltaX += deltaX;
     this.totalDeltaY += deltaY;
 
+    const initialSelectionAABB = this._getOrCreateSelectionAABB(instances);
+
+    let roundedTotalDeltaX;
+    let roundedTotalDeltaY;
+    if (this.options.snap && this.options.grid) {
+      // round the grabbed node position on the grid
+      const grabbingRelativePosition =
+        resizeGrabbingRelativePositions[grabbingLocation];
+      const initialGrabbingX =
+        initialSelectionAABB.left +
+        initialSelectionAABB.width() * grabbingRelativePosition[0];
+      const initialGrabbingY =
+        initialSelectionAABB.top +
+        initialSelectionAABB.height() * grabbingRelativePosition[1];
+      const grabbingPosition = this._temporaryGrabbingPosition;
+      grabbingPosition[0] = initialGrabbingX + this.totalDeltaX;
+      grabbingPosition[1] = initialGrabbingY + this.totalDeltaY;
+      roundPositionForResizing(
+        grabbingPosition,
+        this.options.gridWidth,
+        this.options.gridHeight,
+        this.options.gridOffsetX,
+        this.options.gridOffsetY,
+        this.options.gridType
+      );
+      roundedTotalDeltaX = grabbingPosition[0] - initialGrabbingX;
+      roundedTotalDeltaY = grabbingPosition[1] - initialGrabbingY;
+    } else {
+      roundedTotalDeltaX = this.totalDeltaX;
+      roundedTotalDeltaY = this.totalDeltaY;
+    }
+    if (!isFreeOnX(grabbingLocation)) {
+      roundedTotalDeltaX = 0;
+    }
+    if (!isFreeOnY(grabbingLocation)) {
+      roundedTotalDeltaY = 0;
+    }
+
+    const isLeft =
+      grabbingLocation === 'TopLeft' ||
+      grabbingLocation === 'BottomLeft' ||
+      grabbingLocation === 'Left';
+    const isTop =
+      grabbingLocation === 'TopLeft' ||
+      grabbingLocation === 'TopRight' ||
+      grabbingLocation === 'Top';
+
+    const flippedTotalDeltaX = isLeft
```

Comment:
"signed"?

Reply:
(rather than flipped?)

Reply:
Actually, the roundedTotalDeltaX can be negative and so do flippedTotalDeltaX.
This is more like a symmetry to end up in the nominal case of a bottom-left grabbed node.
I will add a comment.